### PR TITLE
[#11433] Add option for removing unregistered instructors

### LINKED ISSUE #11433: Admin: Give a way to remove unregistered instructors and their data
Suggestion: Give a way to remove unregistered instructors.
Reason: Some people can request for a TEAMMATES account and change their mind after receiving the join link. We want to be able to remove such people from the system entirely (i.e., any data entities created for them should be removed).

### PR DESCRIPTION
**Pull request by tjtanjin**
Fixes #11433 

**Outline of Solution**

- Provide a button to delete unregistered instructors when they turn up in search results.
- If the delete button is pressed, a prompt first pops up to ask for confirmation.
- Upon confirmation of delete, the instructor and its associated demo course/students will be deleted (through a new API call).
- The deleted rows are then removed from the search table results.

Example usage (bob1 is registered while bob2 is unregistered):
https://user-images.githubusercontent.com/43908963/139599454-bfeddd24-2f51-4ec4-a272-93690e989e48.mov

### CODE REVIEW

File: src/main/java/teammates/logic/api/Logic.java [hunk 1]
```diff
@@ -331,6 +331,23 @@ public void deleteInstructorCascade(String courseId, String email) {
         instructorsLogic.deleteInstructorCascade(courseId, email);
     }
 
+    /**
+     * Deletes an unregistered instructor cascade its associated demo course and students.
```
Comment (wkurniawan07) [inline src/main/java/teammates/logic/api/Logic.java:335]: I don't think a part of the code that checks whether the course/instructor is only demo exists.

**Review by wkurniawan07 — CHANGES_REQUESTED** [review]
I find it hard to justify proceeding here because:
- This is a very dangerous implementation, as you're conflating unregistered instructor who requested account (target audience of this feature) and general unregistered instructor (actual audience as of the current implementation).
- There shouldn't be a need to implement a new endpoint, as we only need to delete the demo course (`DELETE /course`). The difficulty comes in determining whether the target instructor is a demo instructor that has not legitimately used the system.
- #11396 will make this implementation obsolete.

**Comment by jianhandev** [comment]
@tjtanjin reminder on this PR, some tests are failing.

**Comment by tjtanjin** [comment]
> I find it hard to justify proceeding here because:
> 
> * This is a very dangerous implementation, as you're conflating unregistered instructor who requested account (target audience of this feature) and general unregistered instructor (actual audience as of the current implementation).
> * There shouldn't be a need to implement a new endpoint, as we only need to delete the demo course (`DELETE /course`). The difficulty comes in determining whether the target instructor is a demo instructor that has not legitimately used the system.
> * #11396 will make this implementation obsolete.

Hello and thank you for the feedback! May I clarify what is the distinction between an unregistered instructor who requested account and a general unregistered instructor? I am under the impression that in both instances, there are no courses attached to them and that it is logically sound to delete them.

**Comment by wkurniawan07** [comment]
You're making use presence of `instructor.googleId` to determine whether an instructor is unregistered or not, which is valid. However you seem to forget that not all unregistered instructors are new instructors who requested accounts; they might very well be other instructors added to a course by the course owner.

**Comment by tjtanjin** [comment]
Okie so to be sure that I am understanding this correctly, there is currently no way to distinguish the 2 groups of unregistered instructors. However, this distinction is necessary so that the admin will not delete the group of unregistered instructors who were added by a course owner?

**Comment by jianhandev** [comment]
@wkurniawan07 since #11396 is still in progress, shall we put this PR on hold first?

**Comment by samuelfangjw** [comment]
Superseded by #11582

**Pull request closed without merging**